File: Python/underautomation/universal_robots/primary_interface/masterboard_digital_io.py
import typing
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import MasterboardDigitalIO as masterboard_digital_io

class MasterboardDigitalIO:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = masterboard_digital_io()
		else:
			self._instance = _internal
	@property
	def value(self) -> int:
		return self._instance.Value
	@property
	def bit_array(self) -> typing.Any:
		return self._instance.BitArray
	@property
	def digital0(self) -> bool:
		return self._instance.Digital0
	@property
	def digital1(self) -> bool:
		return self._instance.Digital1
	@property
	def digital2(self) -> bool:
		return self._instance.Digital2
	@property
	def digital3(self) -> bool:
		return self._instance.Digital3
	@property
	def digital4(self) -> bool:
		return self._instance.Digital4
	@property
	def digital5(self) -> bool:
		return self._instance.Digital5
	@property
	def digital6(self) -> bool:
		return self._instance.Digital6
	@property
	def digital7(self) -> bool:
		return self._instance.Digital7
	@property
	def configurable0(self) -> bool:
		return self._instance.Configurable0
	@property
	def configurable1(self) -> bool:
		return self._instance.Configurable1
	@property
	def configurable2(self) -> bool:
		return self._instance.Configurable2
	@property
	def configurable3(self) -> bool:
		return self._instance.Configurable3
	@property
	def configurable4(self) -> bool:
		return self._instance.Configurable4
	@property
	def configurable5(self) -> bool:
		return self._instance.Configurable5
	@property
	def configurable6(self) -> bool:
		return self._instance.Configurable6
	@property
	def configurable7(self) -> bool:
		return self._instance.Configurable7
	@property
	def tool_digital0(self) -> bool:
		return self._instance.ToolDigital0
	@property
	def tool_digital1(self) -> bool:
		return self._instance.ToolDigital1

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/MasterboardDigitalIO.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System.Collections;

namespace PrimaryInterface {
	/// <summary>
	/// Represents the state of digital I/O pins on the UR controller masterboard,
	/// including standard digital, configurable, and tool digital pins.
	/// </summary>
	public class MasterboardDigitalIO {


		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public override bool Equals(object obj)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public override int GetHashCode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Register value
		/// </summary>
		[PackageDescription("Register value")]
public int Value { get; }

		/// <summary>
		/// Register value seen as a bool array
		/// </summary>
		[PackageDescription("Register value seen as a bool array")]
public BitArray BitArray { get; }

		/// <summary>
		/// State of standard digital I/O pin 0.
		/// </summary>
		public bool Digital0 { get; }

		/// <summary>
		/// State of standard digital I/O pin 1.
		/// </summary>
		public bool Digital1 { get; }

		/// <summary>
		/// State of standard digital I/O pin 2.
		/// </summary>
		public bool Digital2 { get; }

		/// <summary>
		/// State of standard digital I/O pin 3.
		/// </summary>
		public bool Digital3 { get; }

		/// <summary>
		/// State of standard digital I/O pin 4.
		/// </summary>
		public bool Digital4 { get; }

		/// <summary>
		/// State of standard digital I/O pin 5.
		/// </summary>
		public bool Digital5 { get; }

		/// <summary>
		/// State of standard digital I/O pin 6.
		/// </summary>
		public bool Digital6 { get; }

		/// <summary>
		/// State of standard digital I/O pin 7.
		/// </summary>
		public bool Digital7 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 0.
		/// </summary>
		public bool Configurable0 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 1.
		/// </summary>
		public bool Configurable1 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 2.
		/// </summary>
		public bool Configurable2 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 3.
		/// </summary>
		public bool Configurable3 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 4.
		/// </summary>
		public bool Configurable4 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 5.
		/// </summary>
		public bool Configurable5 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 6.
		/// </summary>
		public bool Configurable6 { get; }

		/// <summary>
		/// State of configurable digital I/O pin 7.
		/// </summary>
		public bool Configurable7 { get; }

		/// <summary>
		/// State of tool digital I/O pin 0.
		/// </summary>
		public bool ToolDigital0 { get; }

		/// <summary>
		/// State of tool digital I/O pin 1.
		/// </summary>
		public bool ToolDigital1 { get; }
	}
}
